### ISSUE TITLE
GAMA model for District 4

### ISSUE THREAD

**Issue by agrignard** [CityScope/CSL_HCMC#26]
As discussed with @Hai-Hoang-88 we will implement a first GAMA model for this project.

@doorleyr when do you think the issue will be closed https://github.com/CityScope/CSL_HCMC/issues/3 so that GAMA can wait for data sent by your module in order to instanciate it.

In the mean time, even if we don't have yet the aggregate value, @Hai-Hoang-88 can you have a look at this repo taht give the possibility to plug GAMA and CityScope https://github.com/CityScope/CS_GAMABrix

**Comment by doorleyr** [comment]
I'll hopefully have an endpoint you can read from (eg. ...api/table_table_name/od) by the end of the week.

**Comment by agrignard** [comment]
@doorleyr do not hesitate to notify me (through this channel) as soon as we have something that we can plug

**Comment by doorleyr** [comment]
@agrignard 
A first pass of this is now ready. I set up a new table hcmc_gama for testing and the endpoint you can read from to get the trip demand information is here:
https://cityio.media.mit.edu/api/table/hcmc_gama/od

This is currently live and will update in response to any changes in the GEOGRID. Whenever the hash changes, you can read and make an update:
https://cityio.media.mit.edu/api/table/hcmc_rd/meta/hashes/od

The O-D information is a list of 'agents' with the following attributes:
home_geoid: the home location of the agent. This refers to a polygon in zones.geojson
work_geoid: the work location of the agent. This refers to a polygon in zones.geojson
age: agent's age
naics: naics code of agent's occupation
Mode: predicted mode of transportation

The home and work locations refer to polygons in the zones.geojson file which can be found here:
https://github.com/CityScope/CSL_HCMC/blob/main/outputs/zones.geojson
On initialising the simulation, each agent should be assigned a home and work location inside the appropriate polygon.

In order to allow for simulating trips to and from the rest of the city (outside of District 4) without making the network too huge, I also created a multi-scale road network. This network consists of a detailed road network within district 4 and a coarse road network in the rest of the city. The geojson can be found here:
https://github.com/CityScope/CSL_HCMC/blob/main/outputs/network.geojson

Let me know how this structure works for you and we can iterate if necessary. I though of one potential issue: you mentioned using GamaBrix to build the simulation. However, if I understand correctly, Gama-Brix works by reading updates to the GEOGRIDDATA field. To  read from the /od endpoint instead is then going to require some modifications.

**Comment by agrignard** [comment]
Awesome! Let's see how this can be plugged in the model developed with  @Hai-Hoang-88 (by the way @Hai-Hoang-88 can you commit your model ?)

**Comment by Hai-Hoang-88** [comment]
Thank you @doorleyr and @agrignard. The ABM district 4 was created by Tiep.

Please kindly take a look at his work:
https://github.com/csl-hcmc/Analysis/tree/master/District4/models

**Comment by agrignard** [comment]
@Hai-Hoang-88  you can see here that @doorleyr 's work is pushed on this repo so can we please to the same for the GAMA model (It's not possible and sustainalbe to maintain 2 repo in //)

For the front end (CityScopeJS) it might be another issue and I think there has been a fork for other reason. For the backend I insist that we need to be consistent together or we will loose lot of time and information.

@LAAP can you confirm (or not? in this case we will need to see how to manage this) to the team that we keep this repo as the reference one, at least for the moment?

**Comment by LAAP** [comment]
Hi @agrignard , I confirm that this should be the repo of reference. Where is the issue? Are people using other repo?

**Comment by agrignard** [comment]
@LAAP Yes there is other repo  as mention in the comment just above there is also this github
https://github.com/csl-hcmc/Analysis/tree/master/District4/models 

@Hai-Hoang-88 can you clarify the situation so we knwo on which repo to work and ideally be sure that everything is update HERE (I mean this one https://github.com/CityScope/CSL_HCMC)

**Comment by Hai-Hoang-88** [comment]
For me, I don't mind about placeholder for sour-codes. It will be awesome to be centralized in one place. And Tiep is in charged of commit the GAMA. In Wednesday, we may have a little time to discuss about this management.

**Comment by agrignard** [comment]
Dear all, I was curious to see if there any udpate on this part. According to @Hai-Hoang-88 , Tiep might be working on this part. @Tiem le me know if you have nay question or need any help.
